### ISSUE TITLE
Push notifications for messages

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#3934]
This task is for sending push notifications to users for certain in-app messages. Make the connection between push notifications and addons

Design spec: [https://www.figma.com/file/Disx2dX7tw81NQIN39BrYo/In-app-Messaging?node-id=2%3A531](https://www.figma.com/file/Disx2dX7tw81NQIN39BrYo/In-app-Messaging?node-id=2%3A531|smart-link)

**Acceptance criteria:** 

* Messages configured (at the manifest level) to trigger a push notification, should trigger one on Android and iOS.
* The title of the app should say Mozilla VPN, and the text of the message should be the same one as the ‘subject’ of the message.
* The message should display the Mozilla VPN logo.
* If the message is tapped, the app opens with the message in view.
* Messages can be dismissed in whichever native way is normal (e.g. in Android it’s via swiping).

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/VPN-2517)
┆author: Matt Lichtenstein

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

Are these push notifications in the backend sense? (the Bea project) or in the sense of a notification showing up on in the user’s device? Matt Lichtenstein

**Comment by data-sync-user** [comment]
➤ Matt Lichtenstein commented:

!Screen Shot 2022-08-18 at 4.57.35 PM.png|width=285,height=552!

Going strictly off the designs, it looks like they are OS (device) notifications, but I believe they have to be pushed through Apple servers first, and I don’t believe it has much to do with the push implementation that Bea implemented. 

We have some OS push notifications like this in the app already, but I believe they are all local notifications coming from the app running in the background (i.e. vpn disconnected, server unavailable, etc)

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

Right, so I’m trying to understand what this ticket is about. If we already support push notifications (in the sense of OS notifications that show like in the image), then is this ticket about building the logic that connects addOn messages to the routine that launches such notifications?

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Hey Chris Rooney I am working on this and I realize we do not have the text to put on the notifications settings page with regards to these in-app notifications. I came up with something, but would like to have your input as well. Thanks!

!2022-09-01_18-24.png|width=349,height=80!

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

Hey Bea – I think we might want to push the ‘opt-out’ feature out of scope. The rationale is that currently we are planning on being able to define which messages trigger the push notifications and which ones don’t (see the PRD). Having a user override this generates situations that are not very clear:

  * How do we explain that ‘transactional messages' are not subject to this opt-out? We’d have to mention that this opt-out is only for promotional messages, which assumes that we will send messages otherwise (which is not true either - since we are not planning on using a mobile push notification for the ‘upsell to the bundle’ for example).
  * How do we explain that some messages will not generate a push message even when they are opted-in? e.g. We don’t plan on (nor should) sending a mobile push notification for the ‘What’s new' message.

In other words, I’m not sure it’s a meaningful toggle at this point unless we introduce a lot more granularity on a taxonomy that is not fully defined (we are just starting the journey of using messages). I propose we revisit this as v2. Wdyt?

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Santiago Andrigo thanks for jumping in! Alright, yeah I am fine with it. Users that don’t want notifications can always just turn that off on the system level. 

In this case, scratch my last question Chris Rooney 🙂

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Hey Santiago Andrigo after starting to work on this ticket I found many blockers to implement exactly what is described on the description. Would you mind taking a look at the write up ( https://github.com/mozilla-mobile/mozilla-vpn-client/pull/4327#issue-1360421360 ) I did about this on the PR description for awareness? Thanks!
